<!DOCTYPE html>

<!--
     Copyright 2015 Stephen Parker.

     Licensed under Version 3 of the GPL or any later version
   -->

<html>
    <head>
        <meta charset="UTF-8">
        <title>Islet eQTL explorer</title>
        <meta name="viewport" content="width=device-width,initial-scale=1">
        <link rel="stylesheet" href="css/font-awesome.min.css">
        <link rel="stylesheet" href="css/normalize.css">
        <link rel="stylesheet" href="css/isnp.css">

        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
                (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
                                     m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

            ga('create', 'UA-57854873-1', 'auto');
            ga('send', 'pageview');
        </script>

        <script src="js/d3.v3.js"></script>
        <script src="js/d3-tip.js"></script>
        <script src="js/isnp.js"></script>
        <script src="js/data.r8_20170524.js"></script>
        <script src="js/data.r9_20170524.js"></script>
        <script src="js/data.r8_20170217.js"></script>
        <script src="js/data.r9_20170217.js"></script>
        <script src="js/data.r8_20170210.js"></script>
        <script src="js/data.r9_20170210.js"></script>
    </head>
    <body>
        <div id="mask" class="hidden">
            <div id="dialog">
                <h2 class="header"></h2>
                <div class="content"></div>
            </div>
        </div>
        <header id="banner">
            <h1 id="title"><a href=".">ISLET eQTL EXPLORER</a></h1>
            <nav>
                <ul>
                    <li><a href="http://theparkerlab.org/">THE PARKER LAB</a></li>
                    <li><a href="#about">ABOUT</a></li>
                    <li><a href="#help">HELP</a></li>
                    <li><a href="#authors">AUTHORS</a></li>
                </ul>
            </nav>
        </header>
        <div id="main">
            <div id="plotContainer">
                <section id="control">
                    <h2>Controls</h2>
                    <div>
                        <div>
                            <h4>Chromatin state annotation LD threshold</h4>
                            <select id="dataSelector"></select>
                            <h4>Gene</h4>
                            <div id="searchContainer" class="inputbox">
                                <input type="search" id="search" placeholder="Search for a gene by name or Ensembl ID..."/>
                                <button class="cleaner"><i class="fa fa-times-circle"></i></button>
                            </div>
                            <div>
                                <h4>Footprint</h4>
                                <ul id="footprintFilterList" class="filterList"></ul>
                            </div>
                            <div>
                                <h4>Biotype</h4>
                                <ul id="biotypeFilterList" class="filterList"></ul>
                            </div>
                            <div>
                                <h4><span title="Islet expression specificity index decile">iESI decile</span>&nbsp;<a href="#iesiHelp"><i class="fa fa-info-circle"></i></a></h4>
                                <ul id="iesiFilterList" class="numeric filterList"></ul>
                            </div>
                        </div>
                        <div>
                            <h4>Chromatin state</h4>
                            <ul id="chromatinStateFilterList" class="filterList"></ul>
                        </div>
                    </div>
                    <div class="flexleft">
                        <button id="reset" title="Clear all filters and reset the plot">Reset plot</button>
                        <button id="save" title="Save the current session so you can come back to it later.">Save session</button>
                        <button id="share" title="Share this session with others.">Share session</button>
                    </div>
                </section>
                <section id="plotBox">
                    <h2 id="dotCount"></h2>
                    <div id="plot"></div>
                </section>
            </div>

            <section id="variantDetailsContainer">
                <h2>Variant details</h2>
                <div>
                    <table id="variantDetails">
                        <thead>
                            <tr>
                                <th>Explorer ID</th>
                                <th>Variant</th>
                                <th>Gene</th>
                                <th>Gene name</th>
                                <th>RefSeq ID</th>
                                <th>Position</th>
                                <th>Biotype</th>
                                <th>Chromatin state</th>
                                <th>Footprint</th>
                                <th>-log<sub>10</sub>(p-value)</th>
                                <th>Allele 1 (effect allele) / Allele 2</th>
                                <th>Effect size</th>
                                <th>Actions</th>
                            </tr>
                        </thead>
                        <tbody id="variantDetailsBody">
                            <tr id="variantDetailsInstruction">
                                <td colspan="9"><span class="instruction">Click on a variant in the plot to show its details here.</span></td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </section>

            <section id="savedSessionsContainer">
                <h2>Saved sessions</h2>
                <div>
                    <table id="variantDetails">
                        <thead>
                            <tr>
                                <th>Name</th>
                                <th>LD threshold</th>
                                <th>Gene query</th>
                                <th>Biotypes</th>
                                <th>Chromatin states</th>
                                <th>Footprint</th>
                                <th>iESI</th>
                                <th>Variants</th>
                                <th>Actions</th>
                            </tr>
                        </thead>
                        <tbody id="savedSessionsBody">
                            <tr id="savedSessionsInstruction">
                                <td colspan="7"><span class="instruction">Click the Save button at the top to save your explorer state here.</span></td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </section>

        </div>

        <div id="doc">
            <section id="about">
                <h2>About</h2>
                <div>
                    <p>The data from our 2017 paper <a target="_blank" rel="noopener noreferrer" href="https://dx.doi.org/10.1073/pnas.1621192114">Genetic regulatory signatures underlying islet gene expression and type 2 diabetes</a> includes <span id="variantCount">over 3000</span> variants in islet cis-eQTLs. This tool lets you quickly find those relevant to your research.</p>

                    <p>You can find processed data from the paper <a target="_blank" rel="noopener noreferrer" href="https://theparkerlab.med.umich.edu/data/papers/doi/10.1073/pnas.1621192114/">here</a>, including:</p>
                    <ol>
                        <li>Chromatin state annotations for 31 cell types (see Methods, Chromatin state analyses)</li>
                        <li>ATAC-seq peak calls for two islet samples, skeletal muscle, adipose and GM12878 (see Methods, Open chromatin profiling (ATAC-seq))</li>
                        <li>Transcription factor footprint calls for two islet samples (see Methods, ATAC-seq footprints)</li>
                    </ol>
                </div>
            </section>

            <section id="help">
                <h2>Help</h2>
                <div>
                    <p>If it's not working for you, please let us know: <a class="reporter" href="#">parkerlab-software at umich.edu</a>.</p>
                    <h3>General notes</h3>
                    <p>Most recent browsers should work fine. We've noticed that Firefox, much as we love it, can be a little sluggish
                        relative to the others, so if it's too slow for you, please try another browser. Chrome, Safari and Edge seem
                        to be faster, at least for this.</p>
                    <h3>How to use it</h3>
                    <h4>Filters</h4>
                    <p>At any time, you can reset the plot to its initial state with the <b>Reset plot</b> button.</p>
                    <p>If you're interested in a particular gene, type part of its name or Ensembl ID in the text input labeled <b>Gene</b>.</p>
                    <p>You can filter the variants by chromatin state, by biotype, by iESI (islet expression specificity index), or
                        whether an ATAC-seq footprint was observed at the variant location. Simply click any filter in the
                        list. Hovering over a filter will preview the variants it matches. Clicking a filter again will remove it
                        from the active filters. You can select multiple filters to combine sets of variants. Selecting all filters
                        in a filter category is equivalent to selecting none.</p>
                    <h4>The plot itself</h4>
                    <p>Hovering over any shape in the plot will display a tooltip containing the variant's details.</p>
                    <p>Clicking any shape in the plot will add its details to the table below the plot labeled <b>Variant Details</b>. The <b>Gene</b> column in that table contains links to the Ensembl page for the variant's gene. The <b>Position</b> column
                        contains links to 100kb windows centered on the variants in <a href="http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&amp;hgS_otherUserName=hensley&amp;hgS_otherUserSessionName=20160603_islet_eQTL" target="_blank">this UCSC Genome Browser session</a>. If you click a variant by mistake, you can remove it from the table by clicking the <b>Remove</b> button in the last column.</p>
                    <p>You can zoom the plot with your mouse wheel or trackpad. Zooming in separates the variants, making them easier to
                        click. You can also pan the plot by dragging with the mouse. At any time, you can get back to the starting zoom and
                        position by clicking the <b>Reset plot</b> button.</p>
                    <h4>The data</h4>
                    <h5 id="iesiHelp">Islet expression specificity index</h5>
                    <p>The islet expression specificity index (iESI) is a measure of islet-specific gene expression, with a range from 0 to 1. Here we partition the scores into deciles. A gene with an iESI score in decile 10 is expressed highly and specifically in islets compared to other tissue types.</p>
                    <h4>Printing</h4>
                    <p>You should be able to print the page at any time using your browser's default print dialog. If you notice
                        problems with the output, please let us know.</p>
                    <h4>Sessions</h4>
                    <p>You can save the state of the explorer at any time with the <b>Save</b> button at the top right. In the resulting dialog, enter a name to save the session under, and it will be added to the <b>Sessions</b> table. In that table, you can press <b>Load</b> to load a saved session, or <b>Remove</b> to delete them.</p>

                    <p>At any time you can press the <b>Share</b> button at top right to get a URL to share your current session. It has to be copied to the clipboard manually, for browser security reasons; just press the <b>Copy to clipboard</b> button or use the keyboard: Command-C on Macs, Control-C elsewhere.</p>
                </div>
            </section>

            <section id="history">
                <h2>History</h2>
                <div>
                    <h3>24 May 2017</h3>
                    <ul>
                        <li>Added RefSeq IDs</li>
                        <li>Added effect size</li>
                        <li>Made allele ordering consistent in labels; allele 1 is now always the effect allele.</li>
                    </ul>
                    <h3>17 February 2017</h3>
                    <p>We discovered some duplicate records in the data. This should have had no effect beyond overstated variant counts (3360 total, compared to 3314 after these duplicates were removed, and if you searched for a record with duplicates, e.g. SIX3, you'd see one spot in the plot but the tool would claim it was showing more).</p>
                    <p>We've kept the old data available, labeled 10 Feb 2017, so you can review any sessions you might have created.</p>
                    <p>This was of course found because someone with an interest in a particular variant spotted the discrepancy. We appreciate the help; if anything in the data doesn't make sense to you, please let us know at <a class="reporter" href="#">parkerlab-software at umich.edu</a>.</p>

                    <h3>10 February 2017</h3>
                    <p>Published with session saving and sharing.</p>
                </div>
            </section>

            <section id="authors">
                <h2>Authors</h2>
                <div>
                    <table>
                        <thead>
                            <tr>
                                <th>Name</th>
                                <th>Affiliation</th>
                            </tr>
                        </thead>
                        <tbody>
                            <tr>
                                <td>Arushi Varshney<sup>1</sup></td>
                                <td>
                                    <ul>
                                        <li>Department of Human Genetics, University of Michigan, Ann Arbor, Michigan 48105, USA.</li>
                                    </ul>
                                </td>
                            </tr>


                            <tr>
                                <td>Laura J. Scott<sup>1</sup></td>
                                <td>
                                    <ul>
                                        <li>Department of Biostatistics and Center for Statistical Genetics, University of Michigan, Ann Arbor, Michigan 48109, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Ryan P. Welch<sup>1</sup></td>
                                <td>
                                    <ul>
                                        <li>Department of Biostatistics and Center for Statistical Genetics, University of Michigan, Ann Arbor, Michigan 48109, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Michael R. Erdos<sup>1</sup></td>
                                <td>
                                    <ul>
                                        <li>National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Peter S. Chines</td>
                                <td>
                                    <ul>
                                        <li>National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Narisu Narisu</td>
                                <td>
                                    <ul>
                                        <li>National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Richard D'Oliveira Albanus</td>
                                <td>
                                    <ul>
                                        <li>Department of Computational Medicine &amp; Bioinformatics, University of Michigan,  Ann Arbor, Michigan 48109, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Peter Orchard</td>
                                <td>
                                    <ul>
                                        <li>Department of Computational Medicine &amp; Bioinformatics, University of Michigan,  Ann Arbor, Michigan 48109, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Brooke N. Wolford</td>
                                <td>
                                    <ul>
                                        <li>Department of Computational Medicine &amp; Bioinformatics, University of Michigan,  Ann Arbor, Michigan 48109, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Romy Kursawe</td>
                                <td>
                                    <ul>
                                        <li>The Jackson Laboratory for Genomic Medicine, Farmington, CT 06030, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Swarooparani Vadlamudi</td>
                                <td>
                                    <ul>
                                        <li>Department of Genetics, University of North Carolina, Chapel Hill, North Carolina 27599, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Maren Cannon</td>
                                <td>
                                    <ul>
                                        <li>Department of Genetics, University of North Carolina, Chapel Hill, North Carolina 27599, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>John P. Didion</td>
                                <td>
                                    <ul>
                                        <li>National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>John Hensley</td>
                                <td>
                                    <ul>
                                        <li>Department of Computational Medicine &amp; Bioinformatics, University of Michigan,  Ann Arbor, Michigan 48109, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Anthony Kirilusha</td>
                                <td>
                                    <ul>
                                        <li>National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Lori L. Bonnycastle</td>
                                <td>
                                    <ul>
                                        <li>National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>D. Leland Taylor</td>
                                <td>
                                    <ul>
                                        <li>National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA.</li>
                                        <li>European Bioinformatics Institute, European Molecular Biology Laboratory, Wellcome Trust Genome Campus, Hinxton, Cambridgeshire CB10 1SD, UK.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Richard M. Watanabe</td>
                                <td>
                                    <ul>
                                        <li>Department of Preventive Medicine, University of Southern California Keck School of Medicine, Los Angeles, California 90089, USA.</li>
                                        <li>Department of Physiology and Biophysics, University of Southern California Keck School of Medicine, Los Angeles, California 90089, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Karen L. Mohlke</td>
                                <td>
                                    <ul>
                                        <li>Department of Genetics, University of North Carolina, Chapel Hill, North Carolina 27599, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Michael Boehnke<sup>1</sup></td>
                                <td>
                                    <ul>
                                        <li>Department of Biostatistics and Center for Statistical Genetics, University of Michigan, Ann Arbor, Michigan 48109, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Francis S. Collins<sup>1</sup></td>
                                <td>
                                    <ul>
                                        <li>National Human Genome Research Institute, National Institutes of Health, Bethesda, MD 20892, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Stephen C.J. Parker<sup>1, 2</sup></td>
                                <td>
                                    <ul>
                                        <li>Department of Human Genetics, University of Michigan, Ann Arbor, Michigan 48105, USA.</li>
                                        <li>Department of Computational Medicine & Bioinformatics, University of Michigan, Ann Arbor, Michigan 48109, USA.</li>
                                    </ul>
                                </td>
                            </tr>

                            <tr>
                                <td>Michael L. Stitzel<sup>1</sup></td>
                                <td>
                                    <ul>
                                        <li>The Jackson Laboratory for Genomic Medicine, Farmington, CT 06030, USA.</li>
                                    </ul>
                                </td>
                            </tr>
                        </tbody>
                    </table>
                </div>
                <div class="footnote"><sup>1</sup> Equal contribution <sup>2</sup> Corresponding author: <a class="reporter" href="#">scjp at umich.edu</a></div>
            </section>
        </div>
    </body>
</html>
